test/parallel/test-fs-readSync-optional-params.js

```diff
@@ -5,23 +5,53 @@ const fixtures = require('../common/fixtures');
 const fs = require('fs');
 const assert = require('assert');
 const filepath = fixtures.path('x.txt');
-const fd = fs.openSync(filepath, 'r');
 
 const expected = Buffer.from('xyz\n');
 
 function runTest(defaultBuffer, options) {
-  const result = fs.readSync(fd, defaultBuffer, options);
-  assert.strictEqual(result, expected.length);
-  assert.deepStrictEqual(defaultBuffer, expected);
+  let fd;
+  try {
+    fd = fs.openSync(filepath, 'r');
+    const result = fs.readSync(fd, defaultBuffer, options);
+    assert.strictEqual(result, expected.length);
+    assert.deepStrictEqual(defaultBuffer, expected);
+  } finally {
+    if (fd != null) fs.closeSync(fd);
+  }
 }
 
-// Test passing in an empty options object
-runTest(Buffer.allocUnsafe(expected.length), { position: 0 });
+for (const options of [
 
-// Test not passing in any options object
-runTest(Buffer.allocUnsafe(expected.length));
+  // Test options object
+  { offset: 0 },
+  { length: expected.length },
+  { position: 0 },
+  { offset: 0, length: expected.length },
+  { offset: 0, position: 0 },
+  { length: expected.length, position: 0 },
+  { offset: 0, length: expected.length, position: 0 },
 
-// Test passing in options
-runTest(Buffer.allocUnsafe(expected.length), { offset: 0,
-                                               length: expected.length,
-                                               position: 0 });
+  { offset: null },
+  { position: null },
+  { position: -1 },
+  { position: 0n },
+
+  // Test sane default params
+  {},
+  null,
+  undefined,
+
+  // Test if bad params are interpreted as default (not mandatory)
+  false,
+  true,
+  Infinity,
+  42n,
+  Symbol(),
```

Comment:
Should we add more?
```suggestion
  Symbol(),
  '',
  'a',
  [],
  () => {},
```

Reply:
Unfortunately, no.
I think those should be added, but later in some PR hardening `option` to be as least not of a primitive type. And it should be an assertion for error.

Strings, arrays and functions have `.length` property, so they are silently interpreted as `{ length: 0 }` or even worse.